### ISSUE TITLE
Asm ignores methods of types nested in generic structs

### ISSUE THREAD

**Issue by jnm2** [ashmind/SharpLab#286]
```cs
using System;
using SharpLab.Runtime;

[JitGeneric(typeof(int))] // Removing this line does not cause the normal open generics message
public struct C<T>
{
    public class D
    {
        public void M()
        {
            Console.WriteLine("hi");
        }
    }
}
```

[Results in](https://sharplab.io/#v2:EYLgHgbALANALiAhgZwLYB8ACAGABJgRgG4BYAKB1wGUALRAJwAcAZRYAOgCUBXAOzgCWqAKaky5ANoApAXADiw3sPoCAxgAo4AT0bCA9gDN1A/gEpTAXXKYAzLmRx63VXFwBhADwAVAHzkA3uQAkLb4AEy4ACLkuLG4gWRxSfh2mFC4ALLqpjHJsQl5eYQAnOoARDQCZaZihbgAvrmxjWT1QA==):

```asm
; Desktop CLR v4.7.2563.00 (clr.dll) on amd64.
```

**Comment by ashmind** [comment]
Interesting! I'll take a look.

**Comment by ashmind** [comment]
Fixed by #285.
